test/e2e/apps/controller_revision.go

```diff
@@ -0,0 +1,253 @@
+			updatedControllerRevision, err = csAppsV1.ControllerRevisions(ns).Update(context.TODO(), updatedControllerRevision, metav1.UpdateOptions{})
+			return err
+		})
+		framework.ExpectNoError(err, "failed to update ControllerRevision in namespace: %s", ns)
+		framework.ExpectEqual(updatedControllerRevision.Labels[currentControllerRevision.Name], "updated", "Did not find 'updated' label for this ControllerRevision. Current labels: %v", currentControllerRevision.Labels)
+		framework.Logf("%s has been updated", updatedControllerRevision.Name)
+
+		ginkgo.By("Generate another ControllerRevision by patching the Daemonset")
+		patch := fmt.Sprintf(`{"spec":{"template":{"spec":{"terminationGracePeriodSeconds": %d}}},"updateStrategy":{"type":"RollingUpdate"}}`, 1)
+
+		_, err = c.AppsV1().DaemonSets(ns).Patch(context.TODO(), dsName, types.StrategicMergePatchType, []byte(patch), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "error patching daemon set")
+
+		ginkgo.By("Confirm that there are two ControllerRevisions")
+		err = wait.PollImmediate(controllerRevisionRetryPeriod, controllerRevisionRetryTimeout, checkControllerRevisionListQuantity(f, dsLabelSelector, 2))
+		framework.ExpectNoError(err, "failed to count required ControllerRevisions")
+
+		updatedLabel := map[string]string{updatedControllerRevision.Name: "updated"}
+		updatedLabelSelector := labels.SelectorFromSet(updatedLabel).String()
+
+		ginkgo.By(fmt.Sprintf("Removing a ControllerRevision via 'DeleteCollection' with labelSelector: %q", updatedLabelSelector))
+		err = csAppsV1.ControllerRevisions(ns).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: updatedLabelSelector})
+		framework.ExpectNoError(err, "Failed to delete ControllerRevision: %v", err)
+
+		ginkgo.By("Confirm that there is only one ControllerRevision")
+		err = wait.PollImmediate(controllerRevisionRetryPeriod, controllerRevisionRetryTimeout, checkControllerRevisionListQuantity(f, dsLabelSelector, 1))
+		framework.ExpectNoError(err, "failed to count required ControllerRevisions")
+
+		list, err := csAppsV1.ControllerRevisions(ns).List(context.TODO(), metav1.ListOptions{})
+		framework.ExpectNoError(err, "failed to list ControllerRevision")
+		framework.ExpectEqual(list.Items[0].Revision, int64(3), "failed to find the expected revision for the Controller")
+		framework.Logf("ControllerRevision %q has revision %d", list.Items[0].Name, list.Items[0].Revision)
+	})
+})
+
+func checkControllerRevisionListQuantity(f *framework.Framework, label string, quantity int) func() (bool, error) {
+	return func() (bool, error) {
+		var err error
+
+		framework.Logf("Requesting list of ControllerRevisions to confirm quantity")
+
+		list, err := f.ClientSet.AppsV1().ControllerRevisions(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: label})
+		if err != nil {
+			return false, err
+		}
+
+		if len(list.Items) != quantity {
+			return false, err
+		}
```

Comment:
```suggestion
		if len(list.Items) != quantity {
			return false, nil
		}
```

Reply:
it's a nit but this way we don't have to do the mental exercise :smile:

Reply:
@aojea Fixed. Is there anything blocking [https://github.com/kubernetes/test-infra/pull/26556](https://github.com/kubernetes/test-infra/pull/26556) at the moment? TIA

Reply:
ping @BenTheElder ^^^